haskell-src/Concordium/Genesis/Parameters.hs

```diff
@@ -91,17 +92,54 @@ parseJSONForGCPV1 =
         _tpRewardPeriodLength <- v .: "rewardPeriodLength"
         _tpMintPerPayday <- v .: "mintPerPayday"
         let gcpCooldownParameters = CooldownParametersV1{..}
-            gcpTimeParameters = TimeParametersV1{..}
+            gcpTimeParameters = SomeParam TimeParametersV1{..}
             gcpPoolParameters = PoolParametersV1{..}
             gcpExchangeRates = makeExchangeRates _erEuroPerEnergy _erMicroGTUPerEuro
             _ppPassiveCommissions = CommissionRates{..}
             _ppCommissionBounds = CommissionRanges{..}
         return GenesisChainParameters{..}
 
+-- |Parse 'GenesisChainParameters' from JSON for 'ChainParametersV2'.
+parseJSONForGCPV2 :: Value -> Parser (GenesisChainParameters' 'ChainParametersV2)
+parseJSONForGCPV2 =
+    withObject "GenesisChainParametersV2" $ \v -> do
+        _erEuroPerEnergy <- v .: "euroPerEnergy"
+        _erMicroGTUPerEuro <- v .: "microGTUPerEuro"
```

Comment:
Would it be possible/a good time to replace GTU by CCD at least for the new things we do?

Reply:
Yes. But this is actually just reusing the existing `ExchangeRates` type we have defined. We can change it if we want.. But I'm not so sure we want to do that as it will lead to more "needless" work later on for dependencies. 

I suggest we do this in a subsequent task in order to keep this feature somewhat focused on the actual new consensus protocol. @abizjak can perhaps chime in on his thoughts?

Reply:
We cannot change JSON. I also agree we should not be renaming stuff in this PR.

Reply:
We "can", but what that means is that all consumers of the API now have to support the two versions. It's not hard to do this, but it is work.

Since V1 API is in maintenance mode we are not going to be spending time on this.